lib/ansible/modules/cloud/vmware/vmware_vm_host_drs_rule.py

```diff
@@ -0,0 +1,452 @@
+            if group.name == group_name:
+                if not host_group and isinstance(group, vim.cluster.VmGroup):
+                    obj_name_list = [vm.name for vm in group.vm]
+                    break
+                elif host_group and isinstance(group, vim.cluster.HostGroup):
+                    obj_name_list = [host.name for host in group.host]
+                    break
+
+        return obj_name_list
+
+    def __check_rule_has_changed(self, rule_obj, cluster_obj=None):
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        existing_rule = self.__normalize_vm_host_rule_spec(rule_obj=rule_obj, cluster_obj=cluster_obj)
+
+        # Check if anything has changed
+        if ((existing_rule['rule_enabled'] == self.__enabled) and
+            (existing_rule['rule_mandatory'] == self.__mandatory) and
+            (existing_rule['rule_vm_group_name'] == self.__vm_group_name) and
+            (existing_rule['rule_affine_host_group_name'] == self.__host_group_name or
+             existing_rule['rule_anti_affine_host_group_name'] == self.__host_group_name)):
+
+            return False
+        else:
+            return True
+
+    def create(self):
+        """
+        Function to create a host VM-Host DRS rule if rule does not exist
+        """
+        rule_obj = self.__get_rule_key_by_name(rule_name=self.__rule_name)
+
+        # Check if rule exists
+        if rule_obj:
+
+            operation = 'edit'
+            rule_changed = self.__check_rule_has_changed(rule_obj)
+
+        else:
+            operation = 'add'
+
+        # Check if anything has changed when editing
+        if operation == 'add' or (operation == 'edit' and rule_changed is True):
+
+            rule = vim.cluster.VmHostRuleInfo()
+
+            # Check if already rule exists
+            if rule_obj:
+                # This need to be set in order to edit a existing rule
+                rule.key = rule_obj.key
+
+            rule.enabled = self.__enabled
+            rule.mandatory = self.__mandatory
+            rule.name = self.__rule_name
+
+            if self.__affinity_rule:
+                rule.affineHostGroupName = self.__host_group_name
+            else:
+                rule.antiAffineHostGroupName = self.__host_group_name
+
+            rule.vmGroupName = self.__vm_group_name
+
+            rule_spec = vim.cluster.RuleSpec(info=rule, operation=operation)
+            config_spec = vim.cluster.ConfigSpecEx(rulesSpec=[rule_spec])
+
+            if not self.module.check_mode:
+
+                task = self.__cluster_obj.ReconfigureEx(config_spec, modify=True)
+                wait_for_task(task)
+
+            self.__changed = True
+
+        rule_obj = self.__get_rule_key_by_name(rule_name=self.__rule_name)
+        self.__result = self.__normalize_vm_host_rule_spec(rule_obj)
+
+        self.__msg = "Created DRS rule %s successfully" % (self.__rule_name)
```

Comment:
Change this message when we are editing an existing DRS rule. Right now it is showing "Created ..." even if we are updating the DRS Rule.

Reply:
Ok, no problem - will fix